apps/extension/src/routes/offscreen/window-management.ts

```diff
@@ -0,0 +1,52 @@
+import { WitnessAndBuildRequest } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1alpha1/view_pb';
+import { offscreenClient } from './types';
+import {
+  Action,
+  WitnessData,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/transaction/v1alpha1/transaction_pb';
+
+const sw = self as unknown as ServiceWorkerGlobalScope;
+
+const OFFSCREEN_DOCUMENT_PATH = '/offscreen.html';
+
+export const handleOffscreenAPI = async (
```

Comment:
It feels a bit weird that the those wanting to do the build action, need to call a offscreen-management function. I think it would be better if all this wrapped `sendOffscreenMessage`:
https://github.com/penumbra-zone/web/pull/303/files#diff-a3ef69bb05e35d0d370d757db6a9d3b23d1f9e9c57e9972407c135614af9ffebR62-R70

And then, the caller just needs to be concerned with what function they want to call, aka buildAction().

Reply:
that's much cleaner, modified!